aea/launcher.py

```diff
@@ -0,0 +1,251 @@
+
+        :param agent: agent dir with configuration to start.
+        :return: awaitable task to get result or exception
+        """
+        stop_event = self._manager.Event()
+        task = self._loop.run_in_executor(
+            self._executor_pool, _run_agent, agent_dir, stop_event
+        )
+        self._task_stop_event[task] = stop_event
+        return task
+
+    def _stop_agent(self, agent_dir: PathLike):  # type: ignore # tricky part, cause need to pass str instead of AEA
+        """
+        Stop particular agent.
+
+        :param agent: AEA instance to stop.
+        :return: None
+        """
+        task = self._agent_task[agent_dir]  # type: ignore # tricky part, cause need to pass str instead of AEA
+        stop_event = self._task_stop_event[task]
+        stop_event.set()
+
+
+class BaseLaunchExecutor:
+    """Base launch executor to implement executors."""
+
+    RUNNER_EXECUTOR_CLASS: Optional[Type[BaseAEAExecutor]] = None
+
+    def __init__(self, agent_dirs: List[PathLike]) -> None:
+        """
+        Inin launche executor.
+
+        :param agent_dirs: sequence of agents configuration directories.
+        """
+        self._agent_dirs = agent_dirs
+        self._runner: Optional[BaseAEAExecutor] = None
+        if not self.RUNNER_EXECUTOR_CLASS:
+            raise ValueError("RUNNER_EXECUTOR_CLASS has to be specified!")
+
+    def _make_agents(self) -> Sequence[AEA]:
+        """
+        Load agents from directories.
+
+        :return: sequence of AEA instances.
+        """
+        return [load_agent(agent_dir) for agent_dir in self._agent_dirs]
+
+    def start(self) -> None:
+        """Start agents."""
+        agents = self._make_agents()
+        self._runner = self.RUNNER_EXECUTOR_CLASS(agents)  # type: ignore # pylint: disable=E1102    # checked on init
+        self._runner.start()
+
+    def stop(self) -> None:
+        """Stop agents."""
+        if not self._runner:
+            raise Exception("not started")
+        self._runner.stop()
+
+    @property
+    def is_running(self) -> bool:
+        """
+        Return running state of the launcher executor.
+
+        :return: bool
+        """
+        return bool(self._runner) and self._runner.is_running  # type: ignore  # checked on init
+
+
+class _AsyncLauncherExecutor(BaseLaunchExecutor):
+    """Launcher uses shared asyncio event loop to run multiple agents."""
+
+    RUNNER_EXECUTOR_CLASS = AsyncExecutor
+
+
+class _ThreadLauncherExecutor(BaseLaunchExecutor):
+    """Launcher uses multiple threads to run multiple agents."""
+
+    RUNNER_EXECUTOR_CLASS = ThreadExecutor
+
+
+class _ProcessLauncherExecutor(BaseLaunchExecutor):
+    """Launcher uses subprocesses to run multiple agents."""
+
+    RUNNER_EXECUTOR_CLASS = ProcessExecutor
+
+    def _make_agents(self) -> Sequence[PathLike]:  # type: ignore  # cause need strs for subprocesses
+        """Return sequence of agents dirs."""
+        return list(self._agent_dirs)
```

Comment:
See above; wrong inheritance

Reply:
looking for an extra abstraction to make it inherited in a proper way